contrib/vcloud/benchmarkclient_executor.py

```diff
@@ -0,0 +1,462 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import json
+import logging
+import os
+import shutil
+import subprocess
+import urllib.request
+
+import benchexec.tooladapter
+import benchexec.util
+from . import vcloud_util as vcloudutil
+
+sys.dont_write_bytecode = True  # prevent creation of .pyc files
+
+DEFAULT_CLOUD_TIMELIMIT = 300  # s
+
+DEFAULT_CLOUD_MEMORY_REQUIREMENT = 7000000000  # 7 GB
+DEFAULT_CLOUD_CPUCORE_REQUIREMENT = 2  # one core with hyperthreading
+DEFAULT_CLOUD_CPUMODEL_REQUIREMENT = ""  # empty string matches every model
+
+STOPPED_BY_INTERRUPT = False
+
+_ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__)))
+
+_JustReprocessResults = False
+IVY_JAR_NAME = "ivy-2.5.0.jar"
+
+
+def init(config, benchmark):
+    global _JustReprocessResults
+    _JustReprocessResults = config.reprocessResults
+    tool_locator = benchexec.tooladapter.create_tool_locator(config)
+    benchmark.executable = benchmark.tool.executable(tool_locator)
+    benchmark.tool_version = benchmark.tool.version(benchmark.executable)
+    environment = benchmark.environment()
+    if environment.get("keepEnv", None) or environment.get("additionalEnv", None):
+        sys.exit(
+            "Unsupported environment configuration in tool-info module, "
+            'only "newEnv" is supported by VerifierCloud'
+        )
+
+
+def get_system_info():
+    return None
+
+
+def download_required_jars():
+    # download ivy if needed
+    ivy_path = os.path.join(_ROOT_DIR, "lib", IVY_JAR_NAME)
+    ivy_download_url = "https://www.sosy-lab.org/ivy/org.apache.ivy/ivy/" + IVY_JAR_NAME
+    if not os.path.isfile(ivy_path):
+        # let the process exit if an exception occurs.
+        urllib.request.urlretrieve(ivy_download_url, ivy_path)  # noqa S310
+
+    # prepare command
+    cmd = ["java", "-Divy.default.resolver=Sosy-Lab", "-jar", "lib/" + IVY_JAR_NAME]
+    cmd += ["-settings", "lib/ivysettings.xml"]
+    cmd += ["-dependency", "org.sosy_lab", "vcloud", "0.+"]
+    cmd += ["-confs", "runtime", "-mode", "dynamic", "-refresh"]
+    cmd += ["-retrieve", "lib/vcloud-jars/[artifact](-[classifier]).[ext]"]
+
+    # install cloud and dependencies
+    ant = subprocess.Popen(
```

Comment:
nitpick: not really related to ant anymore :-)

We could also simply use `subprocess.run()` instead of `communicate()` and `wait()`.